miden-proto/Cargo.toml

```diff
@@ -0,0 +1,14 @@
+[package]
+name = "miden-proto"
+version = "0.1.0"
+edition = "2021"
```

Comment:
Would be great to use a layout we use in other Cargo.toml files here and in other files in the repo. The layout we usually use is:
```
name = 
version = 
description = 
authors = ["miden contributors"]
readme = "README.md"
license = "MIT"
repository = "https://github.com/0xPolygonMiden/miden-node"
categories = [] # if applicable
keywords = []
edition = "2021"
rust-version = "1.67"
```

Reply:
added the metadata